src/state/features/application/actions.ts

```diff
@@ -61,13 +62,15 @@ const updateStatus = createAsyncThunk(
 
 const getCompanyDetailsWithAddress = createAsyncThunk(
   `${name}/companyDetailsWithAddress/get`,
-  async (applicationId: string) => {
+  async ({applicationId, dispatch}: {applicationId: string, dispatch: Dispatch}) => {
```

Comment:
as written in the other comment convert this to an RTK Query apiSlice and don't use dispatch inside another redux reducer. We can then handle the error outside of the Redux code as shown in another comment